src/main/java/com/bench/stockmanagement/OrderHandler.java

```diff
@@ -17,34 +21,50 @@
 public class OrderHandler {
     private final OrderReader reader;
     private final OrderMapper mapper;
-    private final OrderService orderService;
+    private final OrderRepository orderRepository;
 
     @Autowired
-    public OrderHandler(OrderReader reader, OrderMapper mapper, OrderService orderService) {
+    public OrderHandler(OrderReader reader, OrderMapper mapper, OrderRepository orderRepository) {
         this.reader = reader;
         this.mapper = mapper;
-        this.orderService = orderService;
+        this.orderRepository = orderRepository;
     }
 
-    public void loadOrder() {
+    //Save orders read from files
+    public Mono<Result> loadOrders() {
         List<Order> orders = reader.readOrder();
-        List<DBOrder> dbOrders = orders.stream().map(mapper::mapOrder).flatMap(List::stream).collect(Collectors.toList());
+        List<DBOrder> dbOrders = orders.stream()
+                .map(mapper::mapOrder)
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
 
-        orderService.saveOrders(dbOrders);
+        return Flux.fromIterable(dbOrders)
+                .doOnNext(orderRepository::saveOrders)
+                .then(Mono.just(Result.SUCCESS))
```

Comment:
Slightly more simple version of this
```suggestion
                .thenReturn(Result.SUCCESS)
```